marlowe-playground-client/src/Marlowe/Linter.purs

```diff
@@ -288,3 +331,145 @@ constantValue (Term (SubValue a b) _) = do
   pure (va - vb)
 
 constantValue _ = Nothing
+
+suggestions :: Boolean -> String -> IRange -> Array CompletionItem
+suggestions stripParens contract range = case parseContract contract of
+  Left _ -> []
+  Right parsedContract ->
+    let
+      (Holes holes) = view _holes (lint parsedContract)
+    in
+      case Map.lookup "monaco_suggestions" holes of
+        Nothing -> []
+        Just s -> case Array.uncons $ Set.toUnfoldable s of
+          Nothing -> []
+          Just { head } -> holeSuggestions stripParens range head
```

Comment:
Another place where you could use do-notation, and whenever you get these nested case statements, I'm all the more in favour.

Reply:
yeah nice, I used `hush` as well to put the parsing in there since we don't do anything with the error.